src/common/components/repository-row/index.js

```diff
@@ -347,14 +347,20 @@ class RepositoryRow extends Component {
             TODO: show 'Loading' when waiting for status?
               and also show 'Never built' when no builds available
           */}
-          { latestBuild &&
+          { latestBuild && snap.snapcraft_data &&
             <BuildStatus
               link={ `/${fullName}/builds/${latestBuild.buildId}`}
-              status={ latestBuild.status }
+              colour={ latestBuild.status }
               statusMessage={ latestBuild.statusMessage }
               dateStarted={ latestBuild.dateStarted }
             />
           }
+          { !snap.snapcraft_data &&
+            <BuildStatus
+              colour="grey"
+              statusMessage="Never built"
+            />
+          }
```

Comment:
Because the build data and snap data are stored in separate data structures in the store, this rather inelegant switching is the best that can be done for now.

It would be nice to redesign the store to unify snap and build data into a single data structure so that build status can be determined by a reducer instead of in the component.